### PR TITLE
[5714] Update from publish all schools that are lead schools

### PR DESCRIPTION
**Pull request by defong**
### Context
Schools and lead schools

### Changes proposed in this pull request
Added ability to Generate data/lead_schools_publish.csv from Publish api
Added ability to Update schools to lead schools from csv data/lead_schools_publish.csv 

### Guidance to review
Follow the readme

### Important business

- [x] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [x] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [x] Do we need to send any updates to DQT as part of the work in this PR?
- [x] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: lib/tasks/schools_data.rake [hunk 1]
```diff
@@ -50,4 +51,48 @@ namespace :schools_data do
...
+    CSV.open(args.output_path || PUBLISH_CSV_PATH, "w+") do |csv|
+      csv << items.first.keys
+      items.each do |hash|
+        csv << hash.values
+      end
+    end
+  end
+
+  desc "Update schools to lead schools from csv data/lead_schools_publish.csv"
+  task :update_school_to_lead_school_publish, %i[input_path] => [:environment] do |_, args|
```
Comment (darokel) [inline lib/tasks/schools_data.rake:82]: could we just rename this to `update_lead_schools_from_publish`?

File: lib/tasks/schools_data.rake [hunk 2]
```diff
@@ -50,4 +51,48 @@ namespace :schools_data do
 
     puts "Done! created: #{created}, updated: #{updated}"
   end
+
+  desc "Generate data/lead_schools_publish.csv from Publish api"
+  task :generate_csv_from_publish, %i[output_path] => [:environment] do |_, args|
```
Comment (darokel) [inline lib/tasks/schools_data.rake:56]: `get_lead_schools_from_publish`? the description already says what it will create

**Review by darokel — COMMENTED** [review]
Looks good so far, just left a couple of comments on name suggestions for the rake tasks

File: lib/tasks/schools_data.rake (hunk 2 above)
Comment (defong) [reply lib/tasks/schools_data.rake]: Done

File: lib/tasks/schools_data.rake (hunk 1 above)
Comment (defong) [reply lib/tasks/schools_data.rake]: Done

**Review by darokel — APPROVED** [review]

**Pull request merged**